src/main/java/http_server/RequestParser.java

```diff
@@ -0,0 +1,82 @@
+package http_server;
+
+import java.util.HashMap;
+
+public class RequestParser {
+    HashMap<String, String> requestCollection = new HashMap<>();
+    HashMap<String, String> headersCollection = new HashMap<>();
+
+    public RequestParser(String request) {
+        String[] splitRequest = request.trim().split("\r\n");
+
+        responseLineParser(splitRequest);
```

Comment:
This would be a cool spot for pulling things into composed objects:

```
requestCollection.putAll(new ResponseLineParser(splitRequest).parse());
requestCollection.putAll(new BodyParser(splitRequest).parse());
headersCollection.putAll(new HeaderParser(splitRequest).parse());
``` 

Sort of in the vein of what I said above, though, I think I only put 👀 on these scenarios because of the whole idea of keeping collaborating lines of code close.  I think the refactor-to-objects is unnecessary, but it does illustrate something interesting: the three lines of code here seem similar at first glance, but are actually marshaling two different bits of object state.   Making that sort of nuance obvious to a reader pays off really frequently